src/pages/main/collaboration/stream/stream.ts

```diff
@@ -0,0 +1,54 @@
+import { CollabReducerAction, Message, MessageDto } from '../model';
+import { useCallback, useEffect, useRef } from 'react';
+import { AGARTEX_COLLABORATION_URL } from '@constants';
+import { ReadyState } from 'react-use-websocket';
+import { useWebSocket } from 'react-use-websocket/dist/lib/use-websocket';
+
+const convertRecordToMap = (record: Record<string, string | null>): Map<string, string | null> => {
+  const map = new Map();
+  Array.from(Object.entries(record)).forEach(([key, val]) => {
+    map.set(key, val);
+  });
+  return map;
+};
+
+const convertMessageDtoToMessage = (data: MessageDto): Message => {
+  if (data.cursorsPositions) {
+    return {
+      ...data,
+      cursorsPositions: convertRecordToMap(data.cursorsPositions)
+    } as Message;
+  }
+  return data as Message;
+};
+
+export const useCollabStream = (dispatch: (action: CollabReducerAction) => void) => {
+  const { 
+    sendMessage: send, 
```

Comment:
`sendJsonMessage` instead maybe, since you do the same things it does later

Reply:
idk `sendJsonMessage` requires `JsonValue` type as argument. It would be more complicated to convert `Message -> JsonValue` that just leave it as it is

Reply:
Ah ok